packages/jest-config/src/parseShardPair.ts

```diff
@@ -0,0 +1,47 @@
+export interface ShardPair {
+  shardCount: number;
+  shardIndex: number;
+}
+
+export const parseShardPair = (pair: string): ShardPair => {
+  const parseResult = parseShardPairResult(pair);
+
+  if (parseResult instanceof Error) {
+    throw parseResult;
+  }
+
+  return parseResult;
+};
+
+export const parseShardPairResult = (pair: string): Error | ShardPair => {
+  const shardPair = pair
+    .split('/')
+    .filter(d => /^\d+$/.test(d))
+    .map(d => parseInt(d, 10))
+    .filter((shard: number) => !Number.isNaN(shard));
+
+  const [shardIndex, shardCount] = shardPair;
+
+  if (shardPair.length !== 2) {
+    throw new Error(
+      'The shard option requires a string in the format of <n>/<m>.',
+    );
+  }
+
+  if (shardIndex === 0 || shardCount === 0) {
```

Comment:
should we validate they're positive numbers as well? quite edge casey, but still

Reply:
Added a new test case to illustrate it does that via the `/\d+/` regex already :)